adagrams/game.py

```diff
@@ -1,11 +1,132 @@
 def draw_letters():
-    pass
+    import random
+    RANGE_LOW = 0
+    RANGE_HIGH = 25
+
+    LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+    list_of_letters = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', "I", "J", "K", \
+                       "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"]
+    hand = []
+    while len(hand) < 10:
+        random_number = random.randint(RANGE_LOW, RANGE_HIGH)
+        letter = list_of_letters[random_number]
+        letter_value = LETTER_POOL.get(letter)
+
+        if letter in hand:
+            n = hand.count(letter)
+            if n < letter_value:
+                hand.append(letter)
+        else:
+            hand.append(letter)
```

Comment:
We've only just started talking about time complexity and Big O, but we have 3 nested loops here (O(n^3)) that we can reduce to one (O(n))! We can use the dictionary `LETTER_POOL` to look up the letter key and decrement the count value.

Because we've got the dictionary created inside the function, that means it'll be rebuilt every time the function is run, and thus we don't need to necessarily "copy" the dictionary to keep a clean copy intact. So we can decrement the count for the letters as we go.

It might look something like this:
```python
while len(hand) < 10:
    # your code to randomly grab a letter
    if LETTER_POOL[letter] > 0:
        hand.append(letter)
        LETTER_POOL[letter] -= 1
```